firstvoices/backend/serializers/page_serializers.py

```diff
@@ -38,3 +47,62 @@ class Meta(SiteContentLinkedTitleSerializer.Meta):
             "banner_image",
             "banner_video",
         )
+
+
+class SitePageDetailWriteSerializer(
+    UpdateSerializerMixin, CreateSiteContentSerializerMixin, SitePageDetailSerializer
+):
+    widgets = serializers.PrimaryKeyRelatedField(
+        queryset=SiteWidget.objects.all(),
+        many=True,
+    )
+    banner_image = serializers.PrimaryKeyRelatedField(
+        queryset=Image.objects.all(),
+        allow_null=True,
+        validators=[SameSite()],
+    )
+    banner_video = serializers.PrimaryKeyRelatedField(
+        queryset=Video.objects.all(),
+        allow_null=True,
+        validators=[SameSite()],
+    )
+
+    def validate_widgets(self, widgets):
+        if not widgets:
+            raise serializers.ValidationError(
+                "The widgets field must contain at least one widget ID."
```

Comment:
Is this too constrained? Would it be useful for people to save a page with just a banner and title, since they can make it team-only?